### PR TITLE
Update version and run tests as part of build process

### PR DESCRIPTION
**Pull request by moble**
The checklist isn't completed because I don't know what the last two items mean.  I followed the instructions to re-render with conda-smithy, even though I don't think it's needed(???), but it led to an error saying `Install conda-forge-pinning or edit conda-forge.yml`.  The project's own [LICENSE file](https://github.com/moble/quaternion/blob/master/LICENSE) is not included in the conda files.

Checklist
* [X] Used a fork of the feedstock to propose changes
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by CJ-Wright** [comment]
`conda install conda-forge-pinning -c conda-forge` should install the pinnings for `conda-smithy`

**Comment by CJ-Wright** [comment]
First don't PR from the feedstock, please create a branch and PR from there. All builds on the feedstock are built and uploaded to anaconda.org

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,8 +28,12 @@ requirements:
     - numpy >=1.13
 
 test:
+  requires:
+    - pytest
   imports:
     - quaternion
+  commands:
+    - pytest
```
Comment (CJ-Wright) [inline recipe/meta.yaml]: If you are going to do this you need to copy the source files which contain the tests over. See: https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#source-files

If the tests take a long time to run I'd advice against this. The tests are just there to make certain that the install is there and that all the deps are installed. Super long testing could prevent other recipes from building or worst case cause the build to fail if the CI times out.

Comment (moble) [reply recipe/meta.yaml]: Yeah, I realized that after the last failure.  The tests take 20-40 seconds all together.

Comment (CJ-Wright) [reply recipe/meta.yaml]: 20-40 seconds are no problem. That statement was aimed at the 30 min test suites.

**Comment by dougalsutherland** [comment]
To add onto what @CJ-Wright was saying: https://conda-forge.org/docs/conda-forge_gotchas.html#using-a-fork-vs-a-branch-when-updating-a-recipe. If you’re going to keep iterating on this, you should probably make a new PR from a fork.

Also, you can ask a bit to rerender for you: https://conda-forge.org/docs/webservice.html#conda-forge-admin-please-rerender

**Pull request closed without merging**